### ISSUE TITLE
Floating point exception with local PQ Overture files

### ISSUE THREAD

**Issue by marklit** [duckdb/duckdb#13504]
### What happens?

Floating point exception while aggregating over Overture's PQ files.

### To Reproduce

```sql
SELECT h3_cell_to_boundary_wkt(
                      h3_latlng_to_cell(bbox.ymax, bbox.xmax, 5))::geometry geom,
                  COUNT(*)
FROM read_parquet('s3://overturemaps-us-west-2/release/2024-08-20.0/theme=addresses/type=address/*.parquet')
WHERE country = 'US'
GROUP BY 1;
```

```
28% ▕████████████████▊                                           ▏ Floating point exception
```

The dataset was produced with Sonoma 14.5.

### OS:

Windows 11 Ubuntu for Windows

### DuckDB Version:

v1.0.0 1f98600c2c

### DuckDB Client:

CLI

### Full Name:

Mark Litwintschik

### Affiliation:

Green Idea

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by szarnyasg** [comment]
@marklit thanks for opening this issue on macOS.

I cannot reproduce this – is this deterministic for you?

```
┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┬──────────────┐
│                                                                                   geom                                                                                    │ count_star() │
│                                                                                 geometry                                                                                  │    int64     │
├───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┼──────────────┤
│ POLYGON ((-112.042118 33.582401, -111.981533 33.663551, -112.03609 33.746633, -112.151306 33.748516, -112.211796 33.66733, -112.157166 33.584297, -112.042118 33.582401)) │       164426 │
│ POLYGON ((-111.7228 32.917532, -111.662308 32.998805, -111.716343 33.082381, -111.830947 33.084633, -111.891348 33.003322, -111.837237 32.919796, -111.7228 32.917532))   │        11133 │
│ POLYGON ((-99.427199 29.899587, -99.361109 29.975301, -99.404041 30.063869, -99.513205 30.076711, -99.579298 30.000937, -99.536225 29.91238, -99.427199 29.899587))       │          744 │
│ POLYGON ((-99.100572 29.860705, -99.034388 29.936216, -99.077 30.024834, -99.185938 30.03793, -99.252126 29.962356, -99.209372 29.873749, -99.100572 29.860705))          │         8935 │
│ POLYGON ((-99.147527 29.532133, -99.081543 29.6078, -99.124081 29.69647, -99.232744 29.709464, -99.298731 29.633735, -99.256053 29.545076, -99.147527 29.532133))         │          317 │
│ POLYGON ((-98.822406 29.492814, -98.756333 29.568278, -98.798551 29.656994, -98.906984 29.670235, -98.973062 29.594708, -98.930702 29.506002, -98.822406 29.492814))      │         6266 │
│ POLYGON ((-98.672112 29.390791, -98.606047 29.466192, -98.648086 29.554941, -98.756333 29.568278, -98.822406 29.492814, -98.780224 29.404075, -98.672112 29.390791))      │       123061 │
│ POLYGON ((-98.497986 29.45276, -98.43183 29.528022, -98.473726 29.616779, -98.581922 29.630265, -98.648086 29.554941, -98.606047 29.466192, -98.497986 29.45276))         │       112233 │
│ POLYGON ((-98.323822 29.514441, -98.257577 29.589562, -98.299329 29.678326, -98.407471 29.691962, -98.473726 29.616779, -98.43183 29.528022, -98.323822 29.514441))       │        84215 │
│ POLYGON ((-98.425013 29.944506, -98.358558 30.019529, -98.400559 30.10821, -98.509161 30.121859, -98.575625 30.046774, -98.533479 29.958101, -98.425013 29.944506))       │         2461 │
│ POLYGON ((-98.575625 30.046774, -98.509161 30.121859, -98.551342 30.210504, -98.660133 30.224055, -98.726605 30.148908, -98.684279 30.060272, -98.575625 30.046774))      │          496 │
│ POLYGON ((-98.400559 30.10821, -98.334004 30.183151, -98.37604 30.271803, -98.484778 30.285506, -98.551342 30.210504, -98.509161 30.121859, -98.400559 30.10821))         │         1373 │
│ POLYGON ((-99.357538 30.392105, -99.291146 30.467578, -99.334193 30.556054, -99.443774 30.569045, -99.510169 30.493512, -99.46698 30.405047, -99.357538 30.392105))       │          102 │
│ POLYGON ((-99.158161 30.618298, -99.09157 30.693545, -99.134512 30.781997, -99.244189 30.795194, -99.310784 30.719887, -99.267699 30.631444, -99.158161 30.618298))       │         2088 │
│ POLYGON ((-98.469429 28.529101, -98.403775 28.604669, -98.445321 28.693552, -98.552662 28.706858, -98.618324 28.631225, -98.576637 28.542352, -98.469429 28.529101))      │          212 │
│ POLYGON ((-97.240403 28.036361, -97.174662 28.111265, -97.214868 28.200278, -97.320959 28.214382, -97.386717 28.139412, -97.346367 28.050404, -97.240403 28.036361))      │          611 │
│ POLYGON ((-97.214868 28.200278, -97.14903 28.27511, -97.189266 28.364113, -97.295485 28.378278, -97.36134 28.30338, -97.320959 28.214382, -97.214868 28.200278))          │         3028 │
│ POLYGON ((-98.049879 29.14548, -97.983755 29.220552, -98.025116 29.309396, -98.132746 29.323162, -98.198881 29.248026, -98.157376 29.159189, -98.049879 29.14548))        │         3809 │
│ POLYGON ((-97.901243 29.042817, -97.835132 29.117825, -97.876315 29.206696, -97.983755 29.220552, -98.049879 29.14548, -98.008551 29.056615, -97.901243 29.042817))       │          644 │
│ POLYGON ((-97.105275 33.00135, -97.036635 33.073996, -97.078454 33.162022, -97.189071 33.177404, -97.257731 33.104702, -97.215754 33.016675, -97.105275 33.00135))        │        53458 │
│                                                                                ·                                                                                          │           ·  │
│                                                                                ·                                                                                          │           ·  │
│                                                                                ·                                                                                          │           ·  │
│ POLYGON ((-77.899942 37.582495, -77.992676 37.535168, -77.982655 37.452819, -77.88023 37.417763, -77.78761 37.46492, -77.7973 37.547302, -77.899942 37.582495))           │         5996 │
│ POLYGON ((-74.052998 39.77761, -74.153811 39.732082, -74.150206 39.646979, -74.046155 39.607398, -73.945512 39.652761, -73.948749 39.73787, -74.052998 39.77761))         │         3037 │
│ POLYGON ((-73.590553 41.470529, -73.69548 41.424511, -73.692597 41.337566, -73.585182 41.296637, -73.480444 41.342489, -73.482932 41.429436, -73.590553 41.470529))       │        22956 │
│ POLYGON ((-71.775295 42.416664, -71.883969 42.371918, -71.88447 42.284061, -71.776707 42.240962, -71.668251 42.285546, -71.66734 42.373391, -71.775295 42.416664))        │        96272 │
│ POLYGON ((-72.654534 43.469492, -72.764958 43.42342, -72.763757 43.334489, -72.652564 43.291634, -72.542359 43.337537, -72.543128 43.426463, -72.654534 43.469492))       │         3881 │
│ POLYGON ((-70.206531 46.31701, -70.326376 46.2724, -70.33014 46.181107, -70.214547 46.13445, -70.094985 46.178897, -70.090734 46.270164, -70.206531 46.31701))            │           24 │
│ POLYGON ((-70.082179 46.452918, -70.202498 46.4084, -70.206531 46.31701, -70.090734 46.270164, -69.9707 46.314519, -69.966179 46.405882, -70.082179 46.452918))           │            1 │
│ POLYGON ((-69.281736 46.030778, -69.401552 45.987253, -69.407187 45.896299, -69.293484 45.848903, -69.173958 45.89227, -69.167846 45.983191, -69.281736 46.030778))       │           20 │
│ POLYGON ((-68.602412 45.744031, -68.721971 45.701325, -68.72895 45.610701, -68.616839 45.562822, -68.497575 45.605375, -68.490127 45.695959, -68.602412 45.744031))       │         3459 │
│ POLYGON ((-67.702625 45.447573, -67.822032 45.405926, -67.830775 45.315695, -67.720572 45.267156, -67.601467 45.308655, -67.592264 45.398841, -67.702625 45.447573))      │          153 │
│ POLYGON ((-67.573746 45.579439, -67.693597 45.537895, -67.702625 45.447573, -67.592264 45.398841, -67.472719 45.440237, -67.463229 45.530512, -67.573746 45.579439))      │           37 │
│ POLYGON ((-68.218374 46.143967, -68.339304 46.101566, -68.347129 46.010667, -68.234501 45.962209, -68.113876 46.004457, -68.105575 46.095315, -68.218374 46.143967))      │          923 │
│ POLYGON ((-68.558415 46.289445, -68.679493 46.246633, -68.686645 46.155563, -68.573201 46.107345, -68.452427 46.150002, -68.444794 46.241032, -68.558415 46.289445))      │           26 │
│ POLYGON ((-67.77778 45.858211, -67.898245 45.81637, -67.906915 45.725778, -67.795588 45.677073, -67.675431 45.718764, -67.666292 45.809311, -67.77778 45.858211))         │          240 │
│ POLYGON ((-67.827591 46.543747, -67.949911 46.501673, -67.95861 46.410511, -67.845473 46.361468, -67.723469 46.403392, -67.714287 46.494509, -67.827591 46.543747))       │          856 │
│ POLYGON ((-69.079558 47.263722, -69.202957 47.220075, -69.20917 47.128164, -69.09249 47.079936, -68.969402 47.123424, -68.962684 47.2153, -69.079558 47.263722))          │          107 │
│ POLYGON ((-68.772484 46.568637, -68.894171 46.525513, -68.900925 46.434188, -68.786481 46.386023, -68.665099 46.42899, -68.657857 46.520278, -68.772484 46.568637))       │            1 │
│ POLYGON ((-68.880578 46.70838, -69.00257 46.6651, -69.009121 46.573649, -68.894171 46.525513, -68.772484 46.568637, -68.765442 46.660051, -68.880578 46.70838))           │           13 │
│ POLYGON ((-67.905868 46.958549, -68.029298 46.916285, -68.037915 46.824781, -67.923595 46.775587, -67.800487 46.8177, -67.791377 46.909158, -67.905868 46.958549))        │         5219 │
│ POLYGON ((-68.118756 47.240551, -68.242831 47.19798, -68.251057 47.106225, -68.135708 47.057083, -68.011955 47.099501, -68.003229 47.191212, -68.118756 47.240551))       │          230 │
├───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┴──────────────┤
│ 17051 rows (40 shown)                                                                                                                                                          2 columns │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

**Comment by marklit** [comment]
It is, I'm been able to produce the issue ~10 times this morning. I shut down my machine and restarted it as well just incase it was the i9-14900K acting up.

**Comment by szarnyasg** [comment]
I started an EC2 `r6i.4xlarge` instance (16 Xeon cores, 128 GB RAM) and ran the queries in issues #13504, #13505, and #13507. The first two completed. #13507 is still running (it didn't crash).

I wonder whether this is a Windows, an Ubuntu or a Windows+Ubuntu (WSL) issue...

**Comment by marklit** [comment]
```
D select extension_version from duckdb_extensions() where extension_name = 'spatial';
┌───────────────────┐
│ extension_version │
│      varchar      │
├───────────────────┤
│ 6826755           │
└───────────────────┘
D force install spatial;
Run Time (s): real 1.492 user 1.715562 sys 0.071681
D select extension_version from duckdb_extensions() where extension_name = 'spatial';
┌───────────────────┐
│ extension_version │
│      varchar      │
├───────────────────┤
│ 6826755           │
└───────────────────┘
```

**Comment by szarnyasg** [comment]
I have re-run the queries on both Windows 11 (vanilla) and Windows 11 + WSL (Ubuntu) and couldn't reproduce the error. This query (#13504) got killed on WSL as the machine only has limited RAM (8GB) but I couldn't observe the FP exception.

I see that @jwass and @blackrez liked the original issues – could others reproduce it?

**Comment by blackrez** [comment]
> I have re-run the queries on both Windows 11 (vanilla) and Windows 11 + WSL (Ubuntu) and couldn't reproduce the error. This query (#13504) got killed on WSL as the machine only has limited RAM (8GB) but I couldn't observe the FP exception.
> 
> I see that @jwass and @blackrez liked the original issues – could others reproduce it?

I remember I get this FP exception on the ouverture maps but I don't have the query anymore (GIS is not my day job, sometimes side project get lost).
I tried the query of @marklit and it get killed, so I filtered with an another country ('FR', we have less address than the US) and it works.

**Comment by marklit** [comment]
It exists without any message each time I run it via cmd.exe and duckdb.exe.

```
C:\Users\markl\Downloads>duckdb.exe
v1.0.0 1f98600c2c
```

```sql
D select extension_version from duckdb_extensions() where extension_name = 'spatial';
┌───────────────────┐
│ extension_version │
│      varchar      │
├───────────────────┤
│ 6826755           │
└───────────────────┘

LOAD h3;
LOAD spatial;
SELECT h3_cell_to_boundary_wkt(
                      h3_latlng_to_cell(bbox.ymax, bbox.xmax, 5))::geometry geom,
                  COUNT(*)
FROM read_parquet('s3://overturemaps-us-west-2/release/2024-08-20.0/theme=addresses/type=address/*.parquet')
WHERE country = 'US'
GROUP BY 1;
 11% ?███████                                                     ?
```

```
C:\Users\markl\Downloads>
```

**Comment by szarnyasg** [comment]
Thanks! This makes me suspect a hardware / network failure... What do you think?

**Comment by marklit** [comment]
I have no idea what might have caused this but my Internet connection is very reliable and I'm using it for 10 other things at the moment. I would point the finger at my CPU but this issue is reliably re-producable on my machine between reboots.

I use DuckDB all day every day and this is the only query where I'm seeing this issue. I have 6-7 cron jobs that use DuckDB to remember state and produce statistics and they run without issue on this machine as frequently as every 5 minutes.

**Comment by prusswan** [comment]
I tried this on Ubuntu 24.04 with no issues. Just fyi, the result has 17051 rows.

**Comment by marklit** [comment]
I'm convinced this is a Windows-specific issue. I'm seeing floating point exceptions with several other datasets I've worked with in other blog posts. When the next stable release of DuckDB comes out, I'll walk through these datasets and see if they're still suffering the same issue and if so, I'll report them and find a way to share them with the team here.

**Comment by carlopi** [comment]
This seems to me connected to https://learn.microsoft.com/en-us/windows/win32/debug/floating-point-exceptions, that article points out supposed to are turned off by default but can be enabled at run-time.

Does this sounds plausible / related to your setup?

**Comment by marklit** [comment]
This issue has now been fixed as of v1.1.1 af39bd0dcf

**Comment by szarnyasg** [comment]
That's fantastic news, thanks for re-running the reproducer.
